addons/sale_mrp/models/account_move.py

```diff
@@ -11,8 +11,13 @@ def _stock_account_get_anglo_saxon_price_unit(self):
 
         so_line = self.sale_line_ids and self.sale_line_ids[-1] or False
         if so_line:
-            bom = self.env['mrp.bom']._bom_find(products=so_line.product_id, company_id=so_line.company_id.id, bom_type='phantom')[so_line.product_id]
-            if bom:
+            # We give preference to the bom in the stock moves for the sale order lines
+            # If there are changes in BOMs between the stock moves creation and the
+            # invoice validation a wrong price will be taken
+            boms = so_line.move_ids.filtered(lambda m: m.state != 'cancel').mapped('bom_line_id.bom_id').filtered(lambda b: b.type == 'phantom')
+            boms2 = (boms or self.env['mrp.bom']._bom_find(products=so_line.product_id, company_id=so_line.company_id.id, bom_type='phantom')[so_line.product_id])
+            if boms2:
+                bom = boms2[:1]
```

Comment:
```suggestion
            boms = so_line.move_ids.filtered(lambda m: m.state != 'cancel').mapped('bom_line_id.bom_id').filtered(lambda b: b.type == 'phantom')
            if boms:
                bom = boms2[0]
```
I don't think creating a kit after the SO confirmation should be taken into account for the bill

Reply:
I agree, I think the boms in the stock moves are just correct.